source/parser/parser.cpp

```
@@ -18,10 +18,37 @@ std::pair<SimulatorVariant, TimeNs> YamlParser::build_simulator_from_config(
         path.parent_path() / parse_topology_config_path(simulation_config);
     const YAML::Node topology_config = YAML::LoadFile(m_topology_config_path);
 
-    process_devices(topology_config);
-    process_links(topology_config);
+    auto parse_if_present =
+        [](const YAML::Node &node,
+           std::function<void(const YAML::Node &)> parser,
+           const char *error_message) {
+            if (node) {
+                parser(node);
+            } else {
+                LOG_ERROR(error_message);
+            }
+        };
+
+    parse_if_present(
+        topology_config["hosts"],
+        [this](auto node) { return process_hosts(node); },
+        "No hosts specified in the topology config");
+
+    parse_if_present(
+        topology_config["switches"],
+        [this](auto node) { return process_switches(node); },
+        "No swithces specified in the topology config");
+
+    parse_if_present(
+        topology_config["links"],
+        [this](auto node) { return process_links(node); },
+        "No swithces specified in the topology config");
```

No links*

---

If there are no nodes, it may be better to throw an exception immediately